marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/EntitySearchControllerTest.java

```diff
@@ -57,30 +71,198 @@ void testCRUDOnSavedQuery() throws Exception {
+            "            \"customerId\",\n" +
+            "            \"name\",\n" +
+            "            \"customerNumber\"\n" +
+            "        ]\n" +
+            "    }\n" +
+            "}";
+
+        int limit = 2;
+        int totalColumns = 3;
+
+        // Export using query document
+        postWithParams(EXPORT_PATH, getRequestParams(limit, json))
+            .andExpect(request().asyncStarted())
+            .andDo(MvcResult::getAsyncResult)
+            .andExpect(status().isOk())
+            .andDo(result -> {
+                String response = result.getResponse().getContentAsString();
+                Set<Integer> actualRowSet = calculateHash(response);
+                assertRowsAndColumns(limit, totalColumns, response);
+                assertTrue(actualRowSet.contains(customer1.getHashCode()));
+                assertTrue(actualRowSet.contains(customer2.getHashCode()));
+            });
+
+        postJson(SAVED_QUERIES_PATH, json)
+            .andExpect(status().isCreated())
+            .andDo(result -> savedQueryResponse = readJsonObject(result.getResponse().getContentAsString()));
+
+        // Export using queryId
+        getJson(EXPORT_PATH + "/query/" + savedQueryResponse.get("savedQuery").get("id").textValue(), getRequestParams(limit, null))
+            .andExpect(request().asyncStarted())
+            .andDo(MvcResult::getAsyncResult)
+            .andExpect(status().isOk())
+            .andDo(result -> {
+                String response = result.getResponse().getContentAsString();
+                Set<Integer> actualRowSet = calculateHash(response);
+                assertRowsAndColumns(limit, totalColumns, response);
+                assertTrue(actualRowSet.contains(customer1.getHashCode()));
+                assertTrue(actualRowSet.contains(customer2.getHashCode()));
+            });
+
+
+        // test limit
+        int newLimit = 1;
+        postWithParams(EXPORT_PATH, getRequestParams(newLimit, json))
+            .andExpect(request().asyncStarted())
+            .andDo(MvcResult::getAsyncResult)
+            .andExpect(status().isOk())
+            .andDo(result -> {
+                String response = result.getResponse().getContentAsString();
+                Set<Integer> actualRowSet = calculateHash(response);
+                assertRowsAndColumns(newLimit, totalColumns, response);
+                assertTrue(actualRowSet.contains(customer1.getHashCode()) || actualRowSet.contains(customer2.getHashCode()));
+            });
+    }
+
+    private void assertRowsAndColumns(int limit, int totalColumns, String response) {
+        int headerRow = 1;
+        int totalRows = limit + headerRow;
+
+        assertEquals(totalRows, response.chars().filter(i -> i == '\n').count() + 1);
+        assertEquals(totalColumns, response.substring(0, response.indexOf("\n")).chars().filter(i -> i == ',').count() + 1);
+    }
+
+    private void createCustomer(Customer customer) {
```

Comment:
This is reusable stuff that is worth adding to the ReferenceModelProject class. 

The current "createCustomer" method in there is for "raw" customers. I think it's worth renaming that to "createRawCustomer", and then this can become "createCustomerInstance", which makes it clear we're creating entity instances here.

Reply:
To the ReferenceModelProject this goes then.